aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -1162,6 +1162,9 @@ void scaled_gemm(
 #ifndef USE_ROCM
   computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_AMAX_D_POINTER, amax_ptr);
   computeDesc.setAttribute(CUBLASLT_MATMUL_DESC_FAST_ACCUM, fastAccuMode);
+#endif
+#if !defined(USE_ROCM)
+  TORCH_CHECK(!isFloat8Type(mat1_dtype) || result_dtype == mat1_dtype, "output dtype must match input dtype for float8 matmuls due to restrictions on amax computation");
```

Comment:
does this disable matmul for non float8 outputs?

I think the matmul should stay enabled, and we just document that amax only makes sense for float8 outputs.  If we really want to signal that it doesn't work perhaps we could populate the amax tensor with a signaling value.

Reply:
for current float8 training recipes we always use float8 inputs and bfloat16 outputs, and we do not care about the amax value.  It would be great if this continued to be supported in cuBLAS.

Reply:
> for current float8 training recipes we always use float8 inputs and bfloat16 outputs, and we do not use the output amax value. It would be great if this continued to be supported in cuBLAS.

Reply:
by the way, @eqy , is there a tracking issue or some other way we can chime in to the discussion of this on the cuBLAS side to advocate for fixing this?  Having only float8 outputs supported in the cuBLAS scaled mm routine would be problematic for float8 use cases.